react-file-uploads/client/src/UploadForm.tsx

```diff
@@ -0,0 +1,83 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { FormEvent, useState } from 'react';
+
+type Image = {
+  imageId: number;
+  url: string;
+  caption: string;
+};
+
+export default function UploadForm() {
+  const [imageFile, setImageFile] = useState();
+  async function handleSubmit(event: FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    /* Prevent the browser's default behavior for form submissions.
+     * Create a `new` FormData object from the `event`.
+     *
+     * Use fetch() to send a POST request to http://localhost:8080/api/uploads.
+     * The body should be the form data object you created (not a JSON string).
+     * Headers are not necessary as fetch will use the correct Content-Type
+     * automatically (multipart/form-data).
+     *
+     * Parse the JSON response body and log the parsed response body.
+     * Set the `imageFile` state to the parsed response body.
+     * Log any errors to the console (using `console.error`).
+     *
+     * References:
+     * https://developer.mozilla.org/en-US/docs/Web/API/FormData/FormData
+     * https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#uploading_a_file
+     */
+
+    try {
+      const formData = new FormData(event.currentTarget);
+      const req = {
+        method: 'POST',
+        body: formData,
+      };
+      const res = await fetch('http://localhost:8080/api/uploads', req);
```

Comment:
Do not use a full path on api calls! Never! (At least, never with the infrastructure we are using.) If you do, you will break the ability to deploy to AWS.